python/lsst/daf/base/propertyContainer/propertyContainerContinued.py

```diff
@@ -441,6 +453,27 @@ def add(self, name, value):
         """
         return _propertyContainerAdd(self, name, value, self._typeMenu)
 
+    def update(self, addition):
+        """Update the current container with the supplied additions.
+
+        Parameters
+        ----------
+        addition : `collections.abc.Mapping` or `PropertySet`
+            The content to merge into the current container.
+
+        Notes
+        -----
+        If the supplied parameter is a `PropertySet` then the
+        `PropertySet.combine` method will be used.  If the supplied parameter
+        is a `collections.abc.Mapping` each item will be copied out of the
+        mapping and value assigned.
+        """
+        if isinstance(addition, PropertySet):
+            self.combine(addition)
+        else:
+            for k, v in addition.items():
+                self[k] = v
```

Comment:
I would be much happier if you omitted this unless you really need it. I think it goes a bit too far in treating PropertySet like a dict.

What does this do if the key already exists? Does this add the new value or replace the existing values? If you must have this then please be much more explicit about its behavior in the documentation.

Reply:
I will add more explanation. I added this because I had a direct need for it.